tests/ui/previews.js

```diff
@@ -0,0 +1,74 @@
+var helpers = require('../lib/helpers');
+
+function testListPreviewsDontExist(test) {
+    test.assertExists('.app-list:not(.expanded)');
+    test.assertNotVisible('.app-list-app .preview');
+}
+
+casper.test.begin('Mobile previews on popular page', {
+    test: function(test) {
+        helpers.startCasper({path: '/popular'});
+
+        helpers.waitForPageLoaded(function() {
+            // Expand listings.
+            casper.click('.app-list-filters-expand-toggle');
+            test.assertVisible('.previews li:first-child img');
+            test.assertNotVisible('.tray .bars');
+            test.assertNotVisible('.tray .arrow-button');
+
+            // Collapse listings.
+            casper.click('.app-list-filters-expand-toggle');
+            testListPreviewsDontExist(test);
+        });
+
+        helpers.done(test);
+    }
+});
+
+casper.test.begin('Desktop previews on popular page', {
+    setUp: helpers.setUpDesktop,
+    tearDown: helpers.tearDown,
+    test: function(test) {
+        helpers.startCasper({path: '/popular'});
+
+        helpers.waitForPageLoaded(function() {
+            // Expand listings.
+            casper.click('.app-list-filters-expand-toggle');
+            test.assertVisible('.previews li:first-child img');
+            test.assertVisible('.tray .bars');
+            test.assertVisible('.tray .arrow-button');
+
+            // Collapse listings.
+            casper.click('.app-list-filters-expand-toggle');
+            testListPreviewsDontExist(test);
+        });
+    }
+});
+
+casper.test.begin('Mobile previews on app detail page', {
+    test: function(test) {
+        helpers.startCasper({path: '/app/something'});
+
+        helpers.waitForPageLoaded(function() {
+            test.assertVisible('.previews .content');
+            test.assertVisible('.content li:first-child img');
+            test.assertNotVisible('.tray .bars');
+            test.assertNotVisible('.tray .arrow-button');
+        });
+    }
+});
+
+casper.test.begin('Desktop previews on app detail page', {
```

Comment:
You can wrap the object with helpers.desktopTest instead of specifying setUp/tearDown manually